bento-sample-app/src/main/java/com/yelp/android/bentosampleapp/ReorderListActivity.kt

```diff
@@ -0,0 +1,107 @@
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_recycler_view)
+
+        addUnorderableListComponent()
+        addLongPressOrderableListComponent()
+        addHandleOrderableListComponent()
+    }
+
+    private fun addUnorderableListComponent() {
+        componentController.addComponent(LabeledComponent("No reorder"))
+        val list = ListComponent<Unit, String>(Unit, LabeledComponentViewHolder::class.java, 2)
+        list.toggleDivider(false)
+        list.setData((0..10).map { it.toString() })
+        componentController.addComponent(list)
+    }
+
+    private fun addLongPressOrderableListComponent() {
+        componentController.addComponent(LabeledComponent("Long press to reorder"))
+        val longPressComponent =
+                ListComponent<Unit, String>(Unit, LabeledComponentViewHolder::class.java, 2)
+        longPressComponent.setIsReorderable(true)
+        longPressComponent.toggleDivider(false)
+        longPressComponent.setData((0..10).map { 'a'.plus(it).toString() })
+        longPressComponent.setOnItemMovedCallback(object : OnItemMovedCallback<String> {
+            override fun onItemMoved(oldIndex: Int, newIndex: Int) {
+                Log.i("Reordered", "Item at $oldIndex moved to $newIndex")
+            }
+        })
+        componentController.addComponent(longPressComponent)
+    }
+
+    private fun addHandleOrderableListComponent() {
+        val handleComponent =
+                ListComponent<Presenter, String>(this, ReorderViewHolder::class.java, 2)
+        handleComponent.setIsReorderable(true)
+        handleComponent.toggleDivider(false)
+        handleComponent.setData((0..10).map { 'A'.plus(it).toString() })
+        handleComponent.setOnItemMovedCallback(object : OnItemMovedCallback<String> {
+            override fun onItemMoved(oldIndex: Int, newIndex: Int) {
+                Log.i("Reordered", "Item at $oldIndex moved to $newIndex")
+            }
+        })
+        val componentGroup = ComponentGroup().apply {
+            addComponent(LabeledComponent("Drag handle to reorder"))
+            addComponent(handleComponent)
+        }
+        componentController.addComponent(componentGroup)
+    }
+
+    override fun onStartDrag(viewHolder: ComponentViewHolder<Presenter, String>) {
+        componentController.onItemPickedUp(viewHolder)
+    }
+}
+
+interface Presenter {
+    fun onStartDrag(viewHolder: ComponentViewHolder<Presenter, String>)
+}
+
+class ReorderViewHolder : ComponentViewHolder<Presenter, String>() {
+
+    private lateinit var text: TextView
+    private lateinit var presenter: Presenter
+
+    override fun inflate(parent: ViewGroup): View {
+        return parent.inflate<View>(R.layout.reorderable_view_holder).also {
+                    text = it.findViewById(R.id.textview)
+                    it.findViewById<View>(R.id.handle).setOnTouchListener { _, event ->
```

Comment:
@dwaxemberg Do you think it would be useful to add some sort of extension function such as `setOnDragStartListener()` that would take a lambda? It feels weird to add a touch listener and listen for `ACTION_DOWN`. It could be used like so:

```
view.setOnDragStartListener {
  presenter.onStartDrag(this)
}

Reply:
Generally I like the idea, but would it be useful to have it in the library vs. making the consumers declare that function if they want it?

Reply:
Spoke on Slack, we agreed it makes sense in the Bento library.